front/src/router/index.ts

```diff
@@ -374,16 +376,29 @@ export function checkPathAvailable (needAuthentication: boolean, needGuest: bool
   return true
 }
 
+/**
+ * Check if the route accessible by roles
+ */
+export function checkRoles (userRole: USER_PROFILE | undefined, requiredRoles: USER_PROFILE[]) {
+  console.log('userRole', userRole)
+  return userRole && requiredRoles.includes(userRole)
+}
+
 router.beforeEach(function (to, from, next) {
   // To handle children routes (to get meta from parents), Vuejs recommend to use to.matched
   // @see: https://github.com/vuejs/vue-router/issues/704
   const needAuthentication = to.matched.some(m => m.meta.needAuthentication)
   appState.hasBurgerMenu = to.matched.some(m => m.meta.hasBurgerMenu)
   const needGuest = to.matched.some(m => m.meta.needGuest)
   const isAuthenticated = authState.data.isAuthenticated
+  console.log(to.matched)
 
   if (!checkPathAvailable(needAuthentication, needGuest, isAuthenticated)) {
     next({ path: isAuthenticated ? ROUTES_PATH.WORKSPACE : ROUTES_PATH.HOME })
+  } else if (
+    to.matched.length > 0 && to.matched[0].meta.requiredRoles && !checkRoles(authState.data?.user?.profile, to.matched[0].meta.requiredRoles)
```

Comment:
It is a bit hard to read, maybe you could create some vars before, like it's done for `checkPathAvailable`.

```ts
const requireProfile = to.matched.some(m => m.meta.requiredRoles)
const authorizedProfiles = [] /* create an array containing all roles authorized to access to this route */
const userProfile = authState.data?.user?.profile
if (!checkPath...) {
} else if (!checkProfile(requireProfile, userProfile, authorizedProfiles)) {
}
```

I change your variable naming, as it seems - to me - more understandable.

User have a profile, and what you're checking is the fact that a route is allowed only for some profiles.
We are not talking about roles, word we use for the role you have as member of a group.

Maybe we need to change this wording if you think this is not easy to understand.

Reply:
I have push a new proposal along these lines. I'll let you have a look